### PR TITLE
chore(liveness): move ws connection start to beginning of liveness flow

### PR DESCRIPTION
**Pull request by thaddmt**
#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .changeset/slimy-items-deny.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@aws-amplify/ui-react-liveness": patch
+"@aws-amplify/ui-react": patch
```
Comment (calebpollman) [inline .changeset/slimy-items-deny.md:3]:
```suggestion
```

**Review by esauerbo — APPROVED** [review]
LGTM! Pulled down the branch and did some smoke testing

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts [hunk 2]
```diff
@@ -223,12 +223,38 @@ export const livenessMachine = createMachine<LivenessContext, LivenessEvent>(
...
+          onDone: {
+            target: 'notRecording',
+            actions: [
+              'updateLivenessStreamProvider',
+              'spawnResponseStreamActor',
+            ],
+          },
+        },
+      },
+      notRecording: {
```
Comment (reesscot) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts:244]: Would it make sense to call this "connecting"? That seems to be what it's called in the e2e test?

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/__tests__/machine.test.ts [hunk 3]
```diff
@@ -381,47 +389,14 @@ describe('Liveness Machine', () => {
...
-      });
-      jest.advanceTimersToNextTimer(); // "recording": "ovalDrawing",
-
-      expect(service.state.value).toEqual({ recording: 'ovalDrawing' });
+      expect(service.state.value).toEqual('start');
     });
   });
 
   describe('recording', () => {
-    it('should reach ovalDrawing state as initial state and respect ovalDrawingTimeout', async () => {
```
Comment (reesscot) [inline packages/react-liveness/src/components/FaceLivenessDetector/service/machine/__tests__/machine.test.ts:400]: Should any of this test be in a new place? I didn't see it move.

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts (hunk 2 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts]: Renamed it already!

File: packages/react-liveness/src/components/FaceLivenessDetector/service/machine/__tests__/machine.test.ts (hunk 3 above)
Comment (thaddmt) [reply packages/react-liveness/src/components/FaceLivenessDetector/service/machine/__tests__/machine.test.ts:400]: I believe it was covered by another test and was not necesary

**Comment by calebpollman** [comment]
Just an fyi, guessing the merge conflicts are due to client utils refactor PR being present on this branch

**Comment by thaddmt** [comment]
> Just an fyi, guessing the merge conflicts are due to client utils refactor PR being present on this branch

Yep just opened a new PR to fix the merge conflicts, was easier this way - https://github.com/aws-amplify/amplify-ui/pull/5234

**Pull request closed without merging**